tests/test_universe.py

```diff
@@ -0,0 +1,413 @@
+        self.assert_allclose([list(p) for p in b.locus_points],
+                             [[0.0, 0.0, 0.0], [0.90625, 0.5625, 0.0],
+                              [2.0, 0.75, 0.0], [3.09375, 0.5625, 0.0],
+                              [4.0, 0.0, 0.0]])
+
+        b.sample(9)
+        self.assertEqual(b.nlocus, 9)
+        self.assertEqual(len(b.locus_points), 9)
+        self.assert_allclose([list(p) for p in b.locus_points],
+                             [[0.0, 0.0, 0.0], [0.41796875, 0.328125, 0.0],
+                              [0.90625, 0.5625, 0.0],
+                              [1.44140625, 0.703125, 0.0], [2.0, 0.75, 0.0],
+                              [2.55859375, 0.703125, 0.0],
+                              [3.09375, 0.5625, 0.0],
+                              [3.58203125, 0.328125, 0.0], [4.0, 0.0, 0.0]])
+
+
+class Bezier3dFp32TC(Bezier3dTB, unittest.TestCase):
+
+    def setUp(self):
+        self.vkls = modmesh.Vector3dFp32
+        self.bkls = modmesh.Bezier3dFp32
+
+    def assert_allclose(self, *args, **kw):
+        if 'rtol' not in kw:
+            kw['rtol'] = 1.e-7
+        return super().assert_allclose(*args, **kw)
+
+
+class Bezier3dFp64TC(Bezier3dTB, unittest.TestCase):
+
+    def setUp(self):
+        self.vkls = modmesh.Vector3dFp32
+        self.bkls = modmesh.Bezier3dFp32
+
+    def assert_allclose(self, *args, **kw):
+        if 'rtol' not in kw:
+            kw['rtol'] = 1.e-15
+        return super().assert_allclose(*args, **kw)
+
+
+class WorldTB(ModMeshTB):
+
+    def test_bezier(self):
+        Vector = self.vkls
+        World = self.wkls
+
+        w = World()
+
+        # Empty
+        self.assertEqual(w.nbezier, 0)
+        with self.assertRaisesRegex(
+                IndexError, "World: \\(bezier\\) i 0 >= size 0"):
+            w.bezier(0)
+
+        # Add Bezier curve
+        b = w.add_bezier(
+            [Vector(0, 0, 0), Vector(1, 1, 0), Vector(3, 1, 0),
+             Vector(4, 0, 0)])
+        self.assertEqual(w.nbezier, 1)
+        with self.assertRaisesRegex(
+                IndexError, "World: \\(bezier\\) i 1 >= size 1"):
+            w.bezier(1)
+
+        # Check control points
+        self.assertEqual(len(b), 4)
+        self.assertEqual(list(b[0]), [0, 0, 0])
+        self.assertEqual(list(b[1]), [1, 1, 0])
+        self.assertEqual(list(b[2]), [3, 1, 0])
+        self.assertEqual(list(b[3]), [4, 0, 0])
+
+        # Check locus points
+        self.assertEqual(b.nlocus, 0)
+        self.assertEqual(len(b.locus_points), 0)
+        b.sample(5)
+        self.assertEqual(b.nlocus, 5)
+        self.assertEqual(len(b.locus_points), 5)
+        self.assert_allclose([list(p) for p in b.locus_points],
+                             [[0.0, 0.0, 0.0], [0.90625, 0.5625, 0.0],
+                              [2.0, 0.75, 0.0], [3.09375, 0.5625, 0.0],
+                              [4.0, 0.0, 0.0]])
+
+        # Confirm we worked on the internal instead of copy
```

Comment:
Checking for getting the internal reference is important for the system to work correctly